apps/pipeline/lib/pipeline/writer/s3_writer/compaction.ex

```diff
@@ -55,9 +56,17 @@ defmodule Pipeline.Writer.S3Writer.Compaction do
     json_table = options[:json_table]
 
     case PrestigeHelper.execute_query("select count(1) from #{json_table}") do
-      {:ok, %{rows: [[0]]}} -> true
-      {:error, _} -> true
-      _ -> false
+      {:ok, %{rows: [[0]]}} ->
+        options[:orc_table]
+        |> TelemetryEventHelper.add_dataset_record_event_count(0)
```

Comment:
We don't care about the records in the json table, right? Especially since there aren't any in this case. We need the record from the main table known here as `:orc_table`